File: services/auth/service-auth/src/main/java/com/mlacker/micros/auth/controller/AccountController.java
package com.mlacker.micros.auth.controller;

import com.mlacker.micros.auth.api.model.AccountModel;
import com.mlacker.micros.auth.service.AccountService;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/accounts")
public class AccountController {

    private final AccountService service;

    public AccountController(AccountService service) {
        this.service = service;
    }

    @GetMapping("{id}")
    public AccountModel get(@PathVariable Long id) {
        return service.get(id);
    }

    @PostMapping
    public void create(@RequestBody AccountModel model) {
        service.create(model);
    }

    @PutMapping
    public void modify(@RequestBody AccountModel model) {
        service.modify(model);
    }

    @DeleteMapping("{id}")
    public void delete(@PathVariable Long id) {
        service.delete(id);
    }
}

File: services/service-auth/service-auth/src/main/kotlin/com/mlacker/micros/auth/controller/AccountController.kt
package com.mlacker.micros.auth.controller

import com.mlacker.micros.auth.api.model.AccountModel
import com.mlacker.micros.auth.domain.user.AccountService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/accounts")
class AccountController(
    private val service: AccountService
) {
    @GetMapping("{id}")
    fun get(@PathVariable id: Long): AccountModel {
        return service[id]
    }

    @PostMapping
    fun create(@RequestBody model: AccountModel) {
        service.create(model)
    }

    @PutMapping
    fun modify(@RequestBody model: AccountModel) {
        service.modify(model)
    }

    @DeleteMapping("{id}")
    fun delete(@PathVariable id: Long) {
        service.delete(id)
    }
}